modules/passkey/contracts/4337/SafeECDSASignerLaunchpad.sol

```diff
@@ -201,85 +348,65 @@ contract SafeECDSASignerLaunchpad is IAccount, SafeStorage {
-        uint256 signerY,
-        address signerVerifier,
-        address setupTo,
-        bytes calldata setupData,
-        address fallbackHandler,
-        bytes memory callData
-    ) external onlySupportedEntryPoint {
-        SafeStorage.singleton = singleton;
-        {
-            address[] memory owners = new address[](1);
-            owners[0] = ICustomECDSASignerFactory(signerFactory).createSigner(signerX, signerY, signerVerifier);
-
-            ISafe(address(this)).setup(owners, 1, setupTo, setupData, fallbackHandler, address(0), 0, payable(address(0)));
-        }
+        bytes32 operationHash = getOperationHash(userOpHash, validAfter, validUntil);
 
-        (bool success, bytes memory returnData) = address(this).delegatecall(callData);
-        if (!success) {
-            // solhint-disable-next-line no-inline-assembly
-            assembly ("memory-safe") {
-                revert(add(returnData, 0x20), mload(returnData))
+        bool failure;
+        if (owners[ICustomECDSASignerFactory(signerFactory).getSigner(signerX, signerY, signerVerifier)] == address(0)) {
+            failure = true;
+        } else {
+            try
+                ICustomECDSASignerFactory(signerFactory).isValidSignatureForSigner(
+                    operationHash,
+                    signature,
+                    signerX,
+                    signerY,
+                    signerVerifier
+                )
+            returns (bytes4 magicValue) {
+                failure = magicValue != ERC1271.MAGIC_VALUE;
+            } catch {
+                failure = true;
             }
         }
 
-        _setInitHash(0);
-    }
-
-    function _domainSeparator() internal view returns (bytes32) {
-        return keccak256(abi.encode(DOMAIN_SEPARATOR_TYPEHASH, block.chainid, SELF));
-    }
-
-    function _getOperationData(
-        bytes32 userOpHash,
-        uint48 validAfter,
-        uint48 validUntil
-    ) internal view returns (bytes memory operationData) {
-        operationData = abi.encodePacked(
-            bytes1(0x19),
-            bytes1(0x01),
-            _domainSeparator(),
-            keccak256(abi.encode(SAFE_INIT_OP_TYPEHASH, userOpHash, validAfter, validUntil, SUPPORTED_ENTRYPOINT))
-        );
+        // The timestamps are validated by the entry point, therefore we will not check them again
+        validationData = _packValidationData(failure, validUntil, validAfter);
     }
 
-    function _initHash() public view returns (bytes32 value) {
+    /**
+     * @notice Reads the configured target Safe singleton address to promote to from storage.
+     * @return value The target Safe singleton address.
+     */
+    function _targetSingleton() internal view returns (address value) {
         // solhint-disable-next-line no-inline-assembly
         assembly ("memory-safe") {
-            value := sload(INIT_HASH_SLOT)
+            value := and(sload(TARGET_SINGLETON_SLOT), 0xffffffffffffffffffffffffffffffffffffffff)
```

Comment:
does it need `and` op here? remember https://github.com/safe-global/safe-smart-account/pull/737?

Reply:
I did this because I don't know if its "legal" to have the `address value` field hold a value that isn't sanitized.

Reply:
So I played with the following contract in remix:
```
// SPDX-License-Identifier: Apache-2.0
pragma solidity ^0.8.0;

contract Test {
    address internal singleton;

    constructor(address _singleton) {
        singleton = _singleton;
    }

    function _targetSingleton() public view returns (address value) {
        // solhint-disable-next-line no-inline-assembly
        assembly ("memory-safe") {
            value := sload(0)
        }
    }
}
```
compiler version 0.8.25, 200 optimizer runs. 

I think the compiler still generates the sanitizing code, and if you have the `and` op, it will execute sanitizing twice (in this particular case, I tested with no optimiser enabled)

Here's the screenshot from debugging the call where the sanitising is executed:
![Screenshot 2024-04-22 at 14 48 31](https://github.com/safe-global/safe-modules/assets/16622558/33d4c785-52a7-4e5b-9fd0-4cdaf2a75a53)

Reply:
Ah, right. It looks like it masks the value on access. Thanks for checking.